Conceptpower+Spring/src/main/java/edu/asu/conceptpower/validation/ConceptTypeAddValidator.java

```diff
@@ -0,0 +1,49 @@
+package edu.asu.conceptpower.validation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+import edu.asu.conceptpower.db4o.TypeDatabaseClient;
+import edu.asu.conceptpower.web.ConceptTypeAddForm;
+
+/**
+ * This class provides methods for validating name, description for concept
+ * Types
+ * 
+ * @author Karthikeyan
+ * 
+ */
+@Component
+public class ConceptTypeAddValidator implements Validator {
+
+    @Autowired
+    private TypeDatabaseClient client;
+
+    @Override
+    public boolean supports(Class<?> clazz) {
+        return clazz.isAssignableFrom(ConceptTypeAddForm.class);
+    }
+
+    @Override
+    public void validate(Object obj, Errors errors) {
+        ValidationUtils.rejectIfEmptyOrWhitespace(errors, "typeName", "required.type_name");
+        ValidationUtils.rejectIfEmptyOrWhitespace(errors, "typeDescription", "required.type_description");
+
+        if (!errors.hasErrors()) {
+            ConceptTypeAddForm conceptTypeAddForm = (ConceptTypeAddForm) obj;
+            // If the name is same as old name then user has not changed the
+            // concept type name
+            if (!conceptTypeAddForm.getOldTypeName().equalsIgnoreCase(conceptTypeAddForm.getTypeName())) {
```

Comment:
@jdamerow I think we discussed about this in the previous pull request comments. This condition was included to make sure duplicate type names are not being added to the system while editing the concept type name. 

Say user first tries to change the concept type name from Test to Test1, the first condition will pass through since 
conceptTypeAddForm.getOldTypeName() which is equal to Test  and 
conceptTypeAddForm.getTypeName() which is equal to Test1.

In the next condition the client.findType(conceptTypeAddForm.getTypeName()) != null , is true and record will be returned for type name Test1 and he will thrown an error on the screen unique type name is required.


Now user changes his mind and does not want to change the Type name and changes the type name to Test (to the old type name that was existing for this type). So in this scenario the first condition will not pass through
!conceptTypeAddForm.getOldTypeName().equalsIgnoreCase(conceptTypeAddForm.getTypeName()).

If the first condition is not available, everytime when the user updates the system by having the same type name, say Test in our scenario he will be thrown an error message "Type name should be unique". This scenario occurs only in editing the concept type names.